### PR TITLE
docs: fully-document readme and improve registry tool

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind cleanup

/kind documentation

**Any specific area of the project related to this PR?**

/area registry

/area documentation

**What this PR does / why we need it**:

This documents the contents of the repo and the plugin registry inside the README. The registry tool has been improved accordingly.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,48 @@
...
 
-This repository contains several reference [plugins](https://deploy-preview-493--falcosecurity.netlify.app/docs/plugins/) that extend the functionality of Falco: The `dummy` and `dummy_c` plugins are used in the [developer's guide](https://deploy-preview-493--falcosecurity.netlify.app/docs/plugins/developers_guide/) to walk through how to write a plugin.
+This repository contains the [Plugin Registry](#plugin-registry) and the [plugins officially maintained](#falcusecurity-plugins) by the Falcosecurity organization. [Plugins](https://deploy-preview-493--falcosecurity.netlify.app/docs/plugins) can be used to extend the functionality of [Falco](https://github.com/falcosecurity/falco) and of applications using [Falcosecurity libs](https://github.com/falcosecurity/libs). Please refer to the [official documentation](https://deploy-preview-493--falcosecurity.netlify.app/docs/plugins/) to better understand the concepts and the architecture behind the plugin system.
```
Comment (leogr) [inline README.md:8]:
```suggestion
This repository contains the [Plugin Registry](#plugin-registry) and the [plugins officially maintained](#falcusecurity-plugins) by the Falcosecurity organization. [Plugins](https://deploy-preview-493--falcosecurity.netlify.app/docs/plugins) can be used to extend [Falco](https://github.com/falcosecurity/falco) and of applications using [Falcosecurity libs](https://github.com/falcosecurity/libs). Please refer to the [official documentation](https://deploy-preview-493--falcosecurity.netlify.app/docs/plugins/) to better understand the plugin system's concepts and architecture. 
```

File: README.md [hunk 2]
```diff
@@ -1,20 +1,48 @@
...
+## Plugin Registry
 
-When ready to release your plugin, make sure to register the plugin with the Falcosecurity organization by creating a PR to modify the [registry.yaml](https://github.com/andreabonanno/plugins/blob/refactor/new-registry-format/registry/registry.yaml) file with details on the new plugin. This ensures that a given ID is used by exactly one source plugin, and allows source plugin authors and extractor plugin authors to coordinate about event source formats.
+The Registry contains metadata and information about every plugin known and recognized by the Falcosecurity organization. These every plugin developed for Falco and made available to the community, including the ones contained in this repository. Check out the sections below to know how to [register your plugins]([registering](#registering-a-new-plugin)), and to see plugins currently contained in the registry.
```
Comment (leogr) [inline README.md:12]:
```suggestion
The Registry contains metadata and information about every plugin known and recognized by the Falcosecurity organization. It lists plugins hosted either in this repository or in other repositories. These plugins are developed for Falco and made available to the community. Check out the sections below to know how to [register your plugins](#registering-a-new-plugin) and see plugins currently contained in the registry.
```

File: README.md [hunk 3]
```diff
@@ -1,20 +1,48 @@
...
 
-# Plugin Registry
+### Registering a new Plugin
 
-<!-- The text inside \<!-- REGISTRY --\> comments is auto-generated. These comments and the text between them should not be edited by hand -->
-<!-- REGISTRY -->
-## Source Plugins
+Registering your plugin inside the registry helps ensuring that some technical constraints are respected, such as that a [given ID is used by exactly one source plugin](https://deploy-preview-493--falcosecurity.netlify.app/docs/plugins/#plugin-event-ids), and allows source plugin authors and extractor plugin authors to [coordinate about event source formats](https://deploy-preview-493--falcosecurity.netlify.app/docs/plugins/#plugin-event-sources-and-interoperability). Moreover, this is a great way to share your plugin project with the community and engage with it, thus gaining new users and **increase its visibility**. We encourage you to register your plugin in this registry before publishing it. You can add your plugins in this registry regardless of where its source code is hosted (there's a `url` field for this specifically).
```
Comment (leogr) [inline README.md:16]:
```suggestion
Registering your plugin inside the registry helps ensure that some technical constraints are respected, such as that a [given ID is used by exactly one source plugin](https://deploy-preview-493--falcosecurity.netlify.app/docs/plugins/#plugin-event-ids) and allows source plugin authors and extractor plugin authors to [coordinate event source formats](https://deploy-preview-493--falcosecurity.netlify.app/docs/plugins/#plugin-event-sources-and-interoperability). Moreover, this is a great way to share your plugin project with the community and engage with it, thus gaining new users and **increasing its visibility**. We encourage you to register your plugin in this registry before publishing it. You can add your plugins in this registry regardless of where its source code is hosted (there's a `url` field for this specifically).
```

File: README.md [hunk 4]
```diff
@@ -1,20 +1,48 @@
...
+
+The registration process consists in adding an entry about your plugin inside the [registry.yaml](./registry.yaml) file by creating a Pull Request in this repository. Please be mindful of few constraints that are automatically checked and required for your plugin to be accepted:
```
Comment (leogr) [inline README.md:18]:
```suggestion
The registration process involves adding an entry about your plugin inside the [registry.yaml](./registry.yaml) file by creating a Pull Request in this repository. Please be mindful of a few constraints that are automatically checked and required for your plugin to be accepted:
```

File: README.md [hunk 5]
```diff
@@ -1,20 +1,48 @@
...
+  contact: https://falco.org/community
+  url: ...
+  license: Apache-2.0
+```
+
+You can find the full registry specification here: *(coming soon...)*
+
+### Registered Plugins
+
+The tables below list the all plugins currently registered. The tables are are automatically generated from [registry.yaml](./registry.yaml).
```
Comment (leogr) [inline README.md:40]:
```suggestion
The tables below list all the plugins currently registered. The tables are automatically generated from the [registry.yaml](./registry.yaml) file.
```

File: README.md [hunk 6]
```diff
@@ -23,10 +51,30 @@ When ready to release your plugin, make sure to register the plugin with the Fal
...
 | Name | Extract Event Sources | Description | Info |
 | --- | --- | --- | --- |
 | [json](https://github.com/falcosecurity/plugins/tree/master/plugins/json) | N/A | Extract values from any JSON payload | Authors: [The Falco Authors](https://falco.org/community) <br/> License: Apache-2.0 |
 
-<!-- REGISTRY -->
+<!-- REGISTRY:EXTRACTOR-TABLE -->
+
+## Hosted Plugins 
+
+Another purpose of this repository is to host and maintain the plugins owned by the Falcosecurity organization. Each plugin is a standalone project and has its own directory, and they are all inside `plugins` folder.
```
Comment (leogr) [inline README.md:66]:
```suggestion
Another purpose of this repository is to host and maintain the plugins owned by the Falcosecurity organization. Each plugin is a standalone project and has its own directory, and they are all inside the [plugins](https://github.com/falcosecurity/plugins/tree/master/plugins) folder.
```

File: README.md [hunk 7]
```diff
@@ -23,10 +51,30 @@ When ready to release your plugin, make sure to register the plugin with the Fal
...
+
+The `master` branch contains the most up-to-date state of development, and each plugin is regularly released. Please check our [Release Process](./release.md) to know about how plugins are released and how artifacts are distributed. Dev builds are published each time a Pull Request gets merged into `master`, whereas stable builds are released and published only when a new releases gets tagged. You can find the published artifacts at https://download.falco.org/?prefix=plugins.
```
Comment (leogr) [inline README.md:68]:
```suggestion
The `master` branch contains the most up-to-date state of development, and each plugin is regularly released. Please check our [Release Process](./release.md) to know how plugins are released and how artifacts are distributed. Dev builds are published each time a Pull Request gets merged into `master`, whereas stable builds are released and published only when a new release gets tagged. You can find the published artifacts at https://download.falco.org/?prefix=plugins.
```

**Review by leogr — COMMENTED** [review]
Great improvement!

I have left some suggestions regarding some issues in the text and others to make the readme more straightforward.

Also, I guess we have to rebuild the tables (and commit the changes directly in this PR).

Otherwise SGTM :)

**Comment by jasondellaluce** [comment]
Thank you @leogr! I applied all your suggestions. The generated table is already up to date in these changes.

**Review by leogr — APPROVED** [review]
/approve

**Pull request merged**